src_new/adapters/prebid.js

```diff
@@ -661,86 +821,283 @@ function fetchBids(activeSlots, impressionID){
-					prebidConfig["schain"] = CONFIG.getSchainObject();
-				}
-				refThis.assignSingleRequestConfigForBidders(prebidConfig);
-				// Adding a hook for publishers to modify the Prebid Config we have generated
-				util.handleHook(CONSTANTS.HOOKS.PREBID_SET_CONFIG, [ prebidConfig ]);
+function addOnBidResponseHandler(){
+	if(util.isFunction(window[pbNameSpace].onEvent)){
+		if(!onEventAdded){
+			window[pbNameSpace].onEvent('bidResponse', refThis.pbBidStreamHandler);
+			onEventAdded = true;
+		}		
+	} else {
+		util.logWarning("PreBid js onEvent method is not available");
+		return;
+	}
+}
+
+exports.addOnBidResponseHandler = addOnBidResponseHandler;
 
-				if(CONFIG.isUserIdModuleEnabled()){
-					prebidConfig["userSync"]["userIds"] = util.getUserIdConfiguration();
+function setPrebidConfig(){
+	if(util.isFunction(window[pbNameSpace].setConfig) || typeof window[pbNameSpace].setConfig == "function") {
+		var prebidConfig = {
+			debug: util.isDebugLogEnabled(),
+			cache: {
+				url: CONSTANTS.CONFIG.CACHE_URL + CONSTANTS.CONFIG.CACHE_PATH
+			},
+			bidderSequence: "random",					
+			disableAjaxTimeout: CONFIG.getDisableAjaxTimeout(),
+			enableSendAllBids: CONFIG.getSendAllBidsStatus(),
+			targetingControls: {
+				alwaysIncludeDeals: true
+			},
+			testGroupId: parseInt(window.PWT.testGroupId || 0)
+		};
+		if(CONFIG.getPriceGranularity()){
+			prebidConfig["priceGranularity"] = CONFIG.getPriceGranularity();
+		}
+
+		if(isPrebidPubMaticAnalyticsEnabled === true){
+			prebidConfig['instreamTracking'] = {
+				enabled: true
+			}
+		}
+
+		refThis.assignUserSyncConfig(prebidConfig);
+		refThis.assignGdprConfigIfRequired(prebidConfig);
+		refThis.assignCcpaConfigIfRequired(prebidConfig);
+		refThis.assignCurrencyConfigIfRequired(prebidConfig);
+		refThis.assignSchainConfigIfRequired(prebidConfig);
+		refThis.assignSingleRequestConfigForBidders(prebidConfig);
+		// Adding a hook for publishers to modify the Prebid Config we have generated
+		util.handleHook(CONSTANTS.HOOKS.PREBID_SET_CONFIG, [ prebidConfig ]);
+		//todo: stop supporting this hook let pubs use pbjs.requestBids hook
+		// do not set any config below this line as we are executing the hook above
+		window[pbNameSpace].setConfig(prebidConfig);
+	} else {
+		util.logWarning("PreBidJS setConfig method is not available");
+	}
+}
+
+exports.setPrebidConfig = setPrebidConfig;
+
+
+function getPbjsAdServerTargetingConfig(){
+	// Todo: Handle send-all bids feature enabled case
+	//		we will need to add bidder specific keys?? do we?
+	// todo: refer constants for key names
+	/*
+		Todo: 
+			what if we do not add a handler for some keys? do we need to add handler to all if we want to add for one?
+			does custom keys do not get used in send-all-bids?
+			do we always need to update the prebid targeting keys config in?
+			what keys in prebid can be re-used?
+	*/
+	return [
+    	//todo: what abt hb_deal, hb_uuid(video?), hb_cache_id(video?), hb_cache_host(video?) ?
```

Comment:
![Codacy](https://codacy.pubmatic.com:443/assets/images/favicon.png) Issue found: [Mixed spaces and tabs.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15196)